lib/rules/no-unused-state.js

```diff
@@ -239,20 +235,29 @@ module.exports = {
         }
       },
 
-      MethodDefinition() {
+      'ClassProperty:exit'(node) {
         if (!classInfo) {
           return;
         }
-        // Create a new set for this.state aliases local to this method.
-        classInfo.aliases = new Set();
+
+        if (
+          !node.static &&
+          node.value &&
+          node.value.type === 'ArrowFunctionExpression'
+        ) {
+          // Remove the current set of aliases local to the previous arrow
+          // function class method
+          classInfo.aliases.clear();
+        }
       },
 
       'MethodDefinition:exit'() {
         if (!classInfo) {
           return;
         }
-        // Forget our set of local aliases.
-        classInfo.aliases = null;
+
+        // Remove the current set of aliases local to the previous class method
+        classInfo.aliases.clear();
```

Comment:
rather than storing across files and clearing; we may want to instead assign `classInfo.aliases = new Set()` at program entry?